CMakeLists.txt

```diff
@@ -106,23 +106,30 @@ include(cmake/SetupBasics.cmake)
 # Find third-party packages
 include(cmake/SetupPackages.cmake)
 
+# Set CUDA compute architecture
 if (RAJA_ENABLE_CUDA)
-  if (DEFINED CMAKE_CUDA_ARCHITECTURES)
-    if ("${CMAKE_CUDA_ARCHITECTURES}" STRLESS "35")
-      message( FATAL_ERROR "RAJA requires minimum CUDA compute architecture of 35")
-    endif()
-  else()
-    message(STATUS "CUDA compute architecture set to RAJA default 35 since it was not specified")
-    set(CMAKE_CUDA_ARCHITECTURES "35" CACHE STRING "Set CMAKE_CUDA_ARCHITECTURES to RAJA minimum supported" FORCE)
-  endif()
+  if (DEFINED CMAKE_CUDA_ARCHITECTURES AND NOT "${CMAKE_CUDA_ARCHITECTURES}" STREQUAL "")
+    foreach (arch IN LISTS CMAKE_CUDA_ARCHITECTURES)
+      if (arch LESS 35)
+        message(FATAL_ERROR
+          "RAJA requires minimum CUDA compute architecture of 35, but got ${arch}")
```

Comment:
I think the following sounds a little better: RAJA requires a minimum CUDA compute architecture of 35, but was given ${arch}.